modules/phase_field/src/postprocessors/FeatureVolumeFraction.C

```diff
@@ -11,70 +11,57 @@
 template<>
 InputParameters validParams<FeatureVolumeFraction>()
 {
-  InputParameters params = validParams<FeatureFloodCount>();
+  InputParameters params = validParams<GeneralPostprocessor>();
+  MooseEnum value_type("VOLUME_FRACTION AVRAMI", "VOLUME_FRACTION");
+  params.addParam<MooseEnum>("value_type", value_type, "The value to output (VOLUME_FRACTION or AVRAMI value)");
   params.addRequiredParam<PostprocessorName>("mesh_volume", "Postprocessor from which to get mesh volume");
-  params.addParam<FileName>("Avrami_file", "filename for Avrami analysis info (ln time and Avrami)");
+  params.addRequiredParam<VectorPostprocessorName>("feature_volumes", "The feature volume VectorPostprocessorValue.");
   params.addParam<Real>("equil_fraction", -1.0, "Equilibrium volume fraction of 2nd phase for Avrami analysis");
-
-  params.set<bool>("calculate_feature_volumes") = true;
   return params;
 }
 
 FeatureVolumeFraction::FeatureVolumeFraction(const InputParameters & parameters) :
-    FeatureFloodCount(parameters),
+    GeneralPostprocessor(parameters),
+    _value_type(getParam<MooseEnum>("value_type").getEnum<ValueType>()),
     _mesh_volume(getPostprocessorValue("mesh_volume")),
-    _equil_fraction(getParam<Real>("equil_fraction"))
+    _feature_volumes(getVectorPostprocessorValue("feature_volumes", "feature_volumes")),
+    _equil_fraction(getParam<Real>("equil_fraction")),
+    _avrami_value(0)
```

Comment:
I chatted a bit with @amjokisaari on this. This was originally her class and she's still using it quite a bit. I'm just trying to maintain the functionality here. Perhaps, I'll see if she'll be willing to rework this or at least be involved in the class changes. We aren't currently testing her new application which may also be affected.